Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,184 @@
+        : QObject()
+        , m_projectInfo(projectInfo)
+    {
+    }
+
+    void ProjectExportWorker::ExportProject()
+    {
+        auto result = ExportProjectForPlatform();
+
+        if (result.IsSuccess())
+        {
+            emit Done();
+        }
+        else
+        {
+            emit Done(result.GetError());
+        }
+    }
+
+    QString ProjectExportWorker::GetLogFilePath() const
+    {
+        QDir logFilePath(m_projectInfo.m_path);
+        // Make directories if they aren't on disk
+        if (!logFilePath.cd(ProjectBuildPathPostfix))
+        {
+            logFilePath.mkpath(ProjectBuildPathPostfix);
+            logFilePath.cd(ProjectBuildPathPostfix);
+
+        }
+        if (!logFilePath.cd(ProjectBuildPathCmakeFiles))
+        {
+            logFilePath.mkpath(ProjectBuildPathCmakeFiles);
+            logFilePath.cd(ProjectBuildPathCmakeFiles);
+        }
+        return logFilePath.filePath(ProjectExportErrorLogName);
+    }
+
+    void ProjectExportWorker::QStringToAZTracePrint([[maybe_unused]] const QString& error)
+    {
+        AZ_TracePrintf("Project Manager", error.toStdString().c_str());
+    }
+
+    //TODO: update this code to just run the desired worker process with the project path argument
+    AZ::Outcome<void, QString> ProjectExportWorker::ExportProjectForPlatform()
+    {
+        // Check if we are trying to cancel task
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        QFile logFile(GetLogFilePath());
+        if (!logFile.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        {
+            QString error = tr("Failed to open log file.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        EngineInfo engineInfo;
+
+        AZ::Outcome<EngineInfo> engineInfoResult = PythonBindingsInterface::Get()->GetEngineInfo();
+        if (engineInfoResult.IsSuccess())
+        {
+            engineInfo = engineInfoResult.GetValue();
+        }
+        else
+        {
+            QString error = tr("Failed to get engine info.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        QTextStream logStream(&logFile);
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            logFile.close();
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        UpdateProgress(tr("Setting Up Environment"));
+
+        auto currentEnvironmentRequest = ProjectUtils::SetupCommandLineProcessEnvironment();
+        if (!currentEnvironmentRequest.IsSuccess())
+        {
+            QStringToAZTracePrint(currentEnvironmentRequest.GetError());
+            return AZ::Failure(currentEnvironmentRequest.GetError());
```

Comment:
You can just return `currentEnvironmentRequest` instead of an `AZ::Failure`. `SetupCommandLineProcessEnvironment` return type is the same `AZ::Outcome<void, QString>`